src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -492,6 +492,11 @@ extern "C" void QCALLTYPE RuntimeMethodHandle_InvokeMethod(
     // If an exception occurs a gc may happen but we are going to dump the stack anyway and we do
     // not need to protect anything.
 
+    // should we use malloc if the size is too big?
+    DWORD objSize = retTH.GetMethodTable()->GetBaseSize();
+    void* localBuf = alloca(objSize);
+    memset(localBuf, 0, objSize);
```

Comment:
I think you also need to GC protect the memory if necessary. It should look like this: https://github.com/dotnet/runtime/blob/a72ec932ee1d51be89d59ae3cdc275e91c8eeed6/src/coreclr/vm/reflectioninvocation.cpp#L548-L563

Reply:
I see, thanks!

Reply:
@jkotas thanks! Do I understand it correctly that we need to define another `FrameWithCookie<ProtectValueClassFrame> *pProtectValueClassFrame = NULL;` (to what we already have) for the ret buff?